src/fetching/tasks/getAllTasks.ts

```diff
@@ -0,0 +1,10 @@
+import { defaultFetch } from '..';
+import { ZetkinTask } from '../../types/zetkin';
+
+export default function getAllTasks(orgId : string, fetch = defaultFetch) {
+    return async () : Promise<ZetkinTask[]> => {
+        const res = await fetch(`/orgs/${orgId}/tasks`);
+        const body = await res.json();
+        return body.data;
```

Comment:
If there is no body, this will return a key error. Better to use `?` here so it returns undefined if no body, rather than thowing an error.

```suggestion
        return body?.data;
```

Reply:
This is a good point and makes me realise that we haven't really talked about how we handle errors --- do we not want it to throw an error if no body? I believe that reactQuery needs an error thrown in order to use it's isError state -- what are your thoughts @richardolsson ?

Reply:
Well we are in fact relying on what @djbusstop pointed out, which is that an error will be thrown (albeit a KeyError), which will cause `isError` (or at least `!isSuccess`) which in turn results in a 404.

But yeah, we should definitely have better error handling in general, and also better loading states – although both of these become more relevant when we do #238 (so that more is loaded client-side where 404 is not a good option).

Reply:
I think one way to make this clearner is have a type `ZetkinPromiseResponse` which is what all zetkin endpoints are expected to return. It would look like this:

```
type ZetkinPromiseResponse<T> = {data: T, error: ZetkinError}
```

Reply:
This could be done in a different PR. Until then, I think it's fine to just remember, be reminded to use `?` on body :) If it's undefined, the place it's used will be able to handle that since it's a normal, valid output of react query anyway.